modules/pricer/src/test/java/com/opengamma/strata/pricer/rate/swaption/HullWhiteSwaptionPhysicalProductPricerTest.java

```diff
@@ -0,0 +1,496 @@
+    double expectedRec = 0.0;
+    double expectedPay = 0.0;
+    for (int loopcf = 0; loopcf < nPayments; loopcf++) {
+      expectedRec += discountedCashFlow[loopcf] * NORMAL.getCDF((kappa + alpha[loopcf]));
+      expectedPay += discountedCashFlow[loopcf] * NORMAL.getCDF(omegaPay * (kappa + alpha[loopcf]));
+    }
+    assertEquals(computedRec.getCurrency(), EUR);
+    assertEquals(computedRec.getAmount(), expectedRec, NOTIONAL * TOL);
+    assertEquals(computedPay.getCurrency(), EUR);
+    assertEquals(computedPay.getAmount(), expectedPay, NOTIONAL * TOL);
+  }
+
+  public void test_presentValue_atMaturity() {
+    CurrencyAmount computedRec =
+        PRICER.presentValue(SWAPTION_REC_LONG, RATES_PROVIDER_AT_MATURITY, HW_PROVIDER_AT_MATURITY);
+    CurrencyAmount computedPay =
+        PRICER.presentValue(SWAPTION_PAY_SHORT, RATES_PROVIDER_AT_MATURITY, HW_PROVIDER_AT_MATURITY);
+    double swapPv = SWAP_PRICER.presentValue(SWAP_REC, RATES_PROVIDER_AT_MATURITY).getAmount(EUR).getAmount();
+    assertEquals(computedRec.getAmount(), swapPv, NOTIONAL * TOL);
+    assertEquals(computedPay.getAmount(), 0d, NOTIONAL * TOL);
+  }
+
+  public void test_presentValue_afterExpiry() {
+    CurrencyAmount computedRec =
+        PRICER.presentValue(SWAPTION_REC_LONG, RATES_PROVIDER_AFTER_MATURITY, HW_PROVIDER_AFTER_MATURITY);
+    CurrencyAmount computedPay =
+        PRICER.presentValue(SWAPTION_PAY_SHORT, RATES_PROVIDER_AFTER_MATURITY, HW_PROVIDER_AFTER_MATURITY);
+    assertEquals(computedRec.getAmount(), 0d, NOTIONAL * TOL);
+    assertEquals(computedPay.getAmount(), 0d, NOTIONAL * TOL);
+  }
+
+  public void test_presentValue_parity() {
+    CurrencyAmount pvRecLong = PRICER.presentValue(SWAPTION_REC_LONG, RATE_PROVIDER, HW_PROVIDER);
+    CurrencyAmount pvRecShort = PRICER.presentValue(SWAPTION_REC_SHORT, RATE_PROVIDER, HW_PROVIDER);
+    CurrencyAmount pvPayLong = PRICER.presentValue(SWAPTION_PAY_LONG, RATE_PROVIDER, HW_PROVIDER);
+    CurrencyAmount pvPayShort = PRICER.presentValue(SWAPTION_PAY_SHORT, RATE_PROVIDER, HW_PROVIDER);
+    assertEquals(pvRecLong.getAmount(), -pvRecShort.getAmount(), NOTIONAL * TOL);
+    assertEquals(pvPayLong.getAmount(), -pvPayShort.getAmount(), NOTIONAL * TOL);
+    double swapPv = SWAP_PRICER.presentValue(SWAP_PAY, RATE_PROVIDER).getAmount(EUR).getAmount();
+    assertEquals(pvPayLong.getAmount() - pvRecLong.getAmount(), swapPv, NOTIONAL * TOL);
+    assertEquals(pvPayShort.getAmount() - pvRecShort.getAmount(), -swapPv, NOTIONAL * TOL);
+  }
+
+  //-------------------------------------------------------------------------
+  public void test_currencyExposure() {
+    MultiCurrencyAmount computedRec = PRICER.currencyExposure(SWAPTION_REC_LONG, RATE_PROVIDER, HW_PROVIDER);
+    MultiCurrencyAmount computedPay = PRICER.currencyExposure(SWAPTION_PAY_SHORT, RATE_PROVIDER, HW_PROVIDER);
+    PointSensitivityBuilder pointRec =
```

Comment:
Why computing the presentValueSensitivity? There is no currency exposure in it for that swap.
Same for next test.